### PR TITLE
HParams: Update the runs table to read columns from redux state

### PR DESCRIPTION
**Pull request by rileyajones**
## Motivation for features / changes
As part the effort to add hparams to time series we have updated the runs table to use the data table in #6394. Now I am updating the runs data table to read its columns from the redux state.

## Screenshots of UI changes (or N/A)
N/A (but there will be soon!)

## Alternate designs / implementations considered (or N/A)
We could have reused the existing sorting property in redux and added keys to it, but I prefered to create a seperate one and remove the existing property when we remove the old runs table.

### CODE REVIEW

File: tensorboard/webapp/runs/store/runs_selectors_test.ts [hunk 1]
```diff
@@ -584,4 +585,66 @@ describe('runs_selectors', () => {
       expect(selectors.getColorGroupRegexString(state)).toEqual('foo(\\d+)');
     });
   });
+
+  describe('#getRunsTableHeaders', () => {
```
[the remark this answers is not in the document]
Comment (rileyajones) [reply tensorboard/webapp/runs/store/runs_selectors_test.ts:589]: That does seem like a good topic for an FE meeting. It feels like a bit of a waste of time to write them but it's been the convention to do so

File: tensorboard/webapp/runs/effects/runs_effects.ts [hunk 2]
```diff
@@ -205,7 +205,7 @@ export class RunsEffects {
       }),
       map((runsAndMedataList) => {
         const newRunsAndMetadata = {} as ExperimentIdToRunsAndMetadata;
-        const runsForAllExperiments = [];
+        const runsForAllExperiments: Run[] = [];
```
[the remark this answers is not in the document]
Comment (rileyajones) [reply tensorboard/webapp/runs/effects/runs_effects.ts:208]: Technically no, I'll revert it.

File: tensorboard/webapp/runs/views/runs_table/runs_table_container.ts [hunk 3]
```diff
@@ -262,22 +272,7 @@ export class RunsTableContainer implements OnInit, OnDestroy {
...
-    },
-  ];
-  sortingInfo = {
-    header: ColumnHeaderType.RUN,
-    name: 'run',
-    order: SortingOrder.ASCENDING,
-  };
-  columnCustomizationEnabled = true;
+  sortingInfo$ = this.store.select(getRunsTableSortingInfo);
   smoothingEnabled = false;
```
[the remark this answers is not in the document]
Comment (rileyajones) [reply tensorboard/webapp/runs/views/runs_table/runs_table_container.ts:276]: I assume you are referring to smoothingEnabled?

File: tensorboard/webapp/testing/utils.ts [hunk 4]
```diff
@@ -81,7 +82,12 @@ export function buildMockState(overrides: Partial<State> = {}): State {
       buildAppRoutingState(overrides[APP_ROUTING_FEATURE_KEY])
     ),
     ...buildStateFromFeatureFlagsState(buildFeatureFlagState()),
-    ...buildStateFromHparamsState(buildHparamsState()),
+    ...buildStateFromHparamsState(
+      buildHparamsState(
```
[the remark this answers is not in the document]
Comment (rileyajones) [reply tensorboard/webapp/testing/utils.ts:86]: When I wrote all these I did it on a single large branch and then broke it apart, this change should have been in #6406 instead.

File: tensorboard/webapp/runs/views/runs_table/runs_table_container.ts [hunk 5]
```diff
@@ -575,18 +582,25 @@ export class RunsTableContainer implements OnInit, OnDestroy {
...
             id: run.id,
             color: colorMap[run.id],
           };
-          this.runsColumns.forEach((column) => {
+          runsColumns.forEach((column) => {
             switch (column.type) {
               case ColumnHeaderType.RUN:
                 tableData[column.name!] = run.name;
                 break;
+              case ColumnHeaderType.HPARAM:
```
[the remark this answers is not in the document]
Comment (rileyajones) [reply tensorboard/webapp/runs/views/runs_table/runs_table_container.ts:598]: Done

File: tensorboard/webapp/runs/views/runs_table/runs_table_container.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (JamesHollyer) [reply tensorboard/webapp/runs/views/runs_table/runs_table_container.ts]: yes

**Review by JamesHollyer — APPROVED** [review]

**Pull request merged**